homeassistant/components/cloud/client.py

```diff
@@ -154,6 +155,28 @@ async def get_google_config(self) -> google_config.CloudGoogleConfig:
 
         return self._google_config
 
+    @callback
+    def async_call_backup_agent_listener(self) -> None:
+        """Register backup agent listener."""
+        if self._backup_agent_listener:
+            self._backup_agent_listener()
+
+    @callback
+    def async_register_backup_agent_listener(
+        self,
+        listener: Callable[[], None],
+    ) -> Callable[[], None]:
+        """Register backup agent listener."""
+
+        @callback
+        def remove_listener() -> None:
+            if self._backup_agent_listener:
+                self._backup_agent_listener()
+                self._backup_agent_listener = None
+
+        self._backup_agent_listener = listener
```

Comment:
Since we override the listener, do you want to raise if a listener is already registered ?

Reply:
Not really, if the backup integration wants to call this several times i don't see why it should raise here instead of getting new cb.

Reply:
Or, just make the `backup_agent_listener` a list, and never have to worry about it 🤷

Reply:
Is there anything to worry about?

Reply:
> Not really, if the backup integration wants to call this several times i don't see why it should raise here instead of getting new cb.

The caller should rely on getting callbacks it asked for until it calls the returned unsubscribe function IMHO.
Overwriting the registered callback function breaks that expectation.